### ISSUE TITLE
Add support for PATCH http requests to servant docs

### ISSUE THREAD

**Issue by timbod7** [haskell-servant/servant#409]
It seems like they are not currently supported:

```
-- | Supported HTTP request methods
data Method = DocDELETE -- ^ the DELETE method
            | DocGET    -- ^ the GET method
            | DocPOST   -- ^ the POST method
            | DocPUT    -- ^ the PUT method
  deriving (Eq, Ord, Generic)
```

**Comment by GarrettMosier** [comment]
Would you please give a specific example of where Patch fails in docs? It isn't clear why you couldn't just add DocPatch to the Method data you posted above.

**Comment by timbod7** [comment]
With the current package as from hackage (0.4.4.6), I see this error:

```
/Users/timd/helix/projects/test/haskell/app/ApiDocs.hs:121:4:
    No instance for (HasLink (Patch '[JSON] ()))
      arising from a use of ‘extraInfo’
    In the expression:
      extraInfo
        (Proxy ::
           Proxy ("users"
                  :>
                  (Capture "userid" UserId
                   :> (ReqBody '[JSON] UserUpdate :> Patch '[JSON] ()))))
```

If I try and hack around this with an HasLink instance:

```
instance HasLink (Patch y r) where
  type MkLink (Patch y r) = URI
  toLink _ = linkURI
```

I get:

```
/Users/timd/helix/projects/test/haskell/app/ApiDocs.hs:121:4:
    No instance for (HasDocs (Patch '[JSON] ()))
      arising from a use of ‘extraInfo’
    In the expression:
      extraInfo
        (Proxy ::
           Proxy ("users"
                  :>
                  (Capture "userid" UserId
                   :> (ReqBody '[JSON] UserUpdate :> Patch '[JSON] ()))))
```

**Comment by GarrettMosier** [comment]
Thanks for that info. I'd be happy to work on it assuming the maintainers would like Patch abilities. Is there a standard way for asking them?

**Comment by soenkehahn** [comment]
I think issues here on github are a good start.

I guess the missing instances for `PATCH` are just an oversight and if you're willing to work on adding those that'd be awesome.

Btw, we have this:

https://github.com/haskell-servant/servant/blob/master/servant/src/Servant/API/Internal/Test/ComprehensiveAPI.hs

It's an API that includes most (unfortunately not all) the combinators we have. It's meant to be used in the different interpretations to test whether all core combinators are supported. You can add a route to `PATCH` there and that should trigger similar errors than the ones mentioned in this issue when executing the test-suite.

Ideally we would have all the verbs included in that api.

(I'm a fan of smaller PRs, so an initial PR that just adds e.g. the HasLink instance would be awesome.)

**Comment by GarrettMosier** [comment]
I'll take it on and see where it goes. I just wanted to make sure that it wasn't intentionally left out because patch isn't used all that often (in my experience). Is there a standard way for grabbing and assigning a ticket?

**Comment by soenkehahn** [comment]
Yes, on the right hand side there is an `Assignee` field. I tried to assign you just now, but that didn't work. But maybe you can assign yourself. (Maybe only people with write access to the repo can be assigned. In that case I would just go without. Should be pretty clear from these comments that you're working on this.)

**Comment by jkarni** [comment]
I would imagine this is already fixed on master. @GarrettMosier can you confirm?

**Comment by GarrettMosier** [comment]
@jkarni I was unable to reproduce the bug.

@timbod7 Were you using the latest version in hackage or the current master branch? If not, please test against the current master branch to confirm that this bug is fixed.

**Comment by timbod7** [comment]
This was using servant (0.4.4.7) from stackage lts. I will check the issue is resolved against the new 0.5 release on hackage

**Comment by jkarni** [comment]
Closing since it seems to be fixed. Please re-open if not.
